uaclient/exceptions.py

```diff
@@ -38,7 +38,7 @@ class NonAutoAttachImageError(UserFacingError):
 class AlreadyAttachedError(UserFacingError):
     """An exception to be raised when a command needs an unattached system."""
 
-    exit_code = 0
+    exit_code = 2
```

Comment:
Note that just changing this exit code will result in breaking the status reported by `systemctl status ua-auto-attach.service` which blindly calls `ua auto-attach` on PRO images every boot. this will the leave the overall `systemctl status` in "State: degraded" which won't look so good.

We need to better handle auto-attach scenario in this PR.

Reply:
I see it, that's what I missed from PRO in the code comments.
I will tackle that, thanks for the pointer